### ISSUE TITLE
DirectoryObjectSecurity should be in the System.Security.AccessControl assembly

### ISSUE THREAD

**Issue by tquerec** [dotnet/corefx#34166]
DirectoryObjectSecurity is only used by the System.DirectoryServices.ActiveDirectorySecurity class to expose Active Directory security descriptors.  It is not used for filesystem ( directory ) descriptors.  It appears it was moved into the System.IO.FileSystem.AccessControl assembly during the port to core when it should have remained in System.Security.AccessControl.

**Comment by bartonjs** [comment]
I agree that it seems the only usage of `DirectoryObjectSecurity` is `ActiveDirectorySecurity` (across .NET Framework and .NET Core).

I just don't know if we have the capability of moving a type "out" (further from System.Object); most of our moves are to move a type "in".

@terrajobst / @ericstj ?

**Comment by ericstj** [comment]
Actually System.Security.AccessControl is closer to System.Object, so this is the normal case of pushing a type down in the stack.

I'm fine with pushing this type into System.Security.AccessControl so long as its doesn't pull anything else down with it or add new dependencies.  If it does we can review those.
